### PR TITLE
Preventive fix: Tensor non-tuple indexing deprecated

### PR DESCRIPTION
**Pull request by nucli-vicky**
In the testing logs right now there is already a warning:

"UserWarning: Using a non-tuple sequence for multidimensional indexing is deprecated and will be changed in pytorch 2.9; use x[tuple(seq)] instead of x[seq]. In pytorch 2.9 this will be interpreted as tensor index, x[torch.tensor(seq)], which will result either in an error or a different result"

This PR is a preventive fix for PyTorch 2.9

### Checks to be done before submitting your PR

- [x] `python3 -m pytest deepinv/tests` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [x] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

**Comment by nucli-vicky** [comment]
Hmm, I might have misinterpreted the warning; it's probably still safe to use lists for indices across a single axis (as is the case for all changes outside of `tensorlist.py`. This likely explains why the warning is only raised for the function `dirac`; apologies..
